### PR TITLE
[4.0] Protect the default workflow

### PR DESCRIPTION
**Pull request by bembelimen**
### Summary of Changes
With this PR the default workflow is not delete- and changeable

### Testing Instructions
- Apply patch
- try to change the default workflow and its stages/transitions 

### Expected result
- not editable
- not deletable

### Actual result
- editable
- deletable

### CODE REVIEW

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql [hunk 1]
```diff
@@ -20,6 +21,7 @@ CREATE TABLE IF NOT EXISTS `#__workflows` (
   KEY `idx_title` (`title`(191)),
   KEY `idx_extension` (`extension`),
   KEY `idx_default` (`default`),
+  KEY `idx_core` (`core`),
```
Comment (csthomas) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql]: Why do you add this index? I do not see any query that will use this column in the WHERE clause.

Comment (bembelimen) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql]: Same for other keys, but perhaps a 3rd party goes directly for the value...

File: administrator/components/com_workflow/Controller/TransitionController.php [hunk 2]
```diff
@@ -90,6 +90,15 @@ protected function allowAdd($data = array())
 	{
 		$user = Factory::getUser();
 
+		$model = $this->getModel('Workflow');
+
+		$workflow = $model->getItem($this->workflowId);
```
Comment (wilsonge) [inline administrator/components/com_workflow/Controller/TransitionController.php:95]: Why do we need this in `allowAdd`

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql (hunk 1 above)
Comment (wilsonge) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql]: I'd kill the index - there's no reason for someone to query this directly

Comment (bembelimen) [reply administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-05-15.sql]: Perhaps not in com _content but probably in other extensions with more than one default workflow?

File: administrator/components/com_workflow/Controller/TransitionController.php (hunk 2 above)
Comment (bembelimen) [reply administrator/components/com_workflow/Controller/TransitionController.php:95]: You're not allowed to change transitions of a core workflow (means you're not allow to add transitions)

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**